### PR TITLE
(ES-2) Create basic homepage

### PR DESCRIPTION
**Pull request by mathieu-valery**
## Ticket
https://anthyou.atlassian.net/browse/ES-2

## Screenshot
image
image

### CODE REVIEW

**Comment by mathieu-valery** [comment]
need clean diff

**Review by AnthYou — CHANGES_REQUESTED** [review]
Following our discussion last week, I let you review and make the discussed changed to this PR accordingly. 🙏

File: escapades_frontend/src/components/Navbar/Navbar.module.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  &__link {
+    @apply fixed left-[66%] top-[-100%] flex-col gap-4 p-4 text-sm w-[100%] bg-gray-800
+            sm:static sm:flex sm:left-auto sm:top-auto sm:flex-row sm:text-base sm:w-auto;
+    transition: 300ms;
+  }
+
+  &__linkActive {
+    @apply bg-white text-gray-800;
+  }
```
Comment (mathieu-valery) [inline escapades_frontend/src/components/Navbar/Navbar.module.scss:13]: here I wanted to use a BEM modifier like `Navbar__link--active` but typescript doesnt allow this syntax..any better idea ?

File: escapades_frontend/src/components/Navbar/Navbar.module.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  &__icons {
+    @apply visible p-2 cursor-pointer
+            sm:hidden sm:p-0;
+  }
+
+  &__open {
+    @apply top-8;
+  }
```
Comment (mathieu-valery) [inline escapades_frontend/src/components/Navbar/Navbar.module.scss:26]: same here

File: escapades_frontend/package.json [hunk 3]
```diff
@@ -11,8 +11,15 @@
     "test": "vitest"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.5.1",
+    "@fortawesome/free-brands-svg-icons": "^6.5.1",
+    "@fortawesome/free-regular-svg-icons": "^6.5.1",
+    "@fortawesome/free-solid-svg-icons": "^6.5.1",
+    "@fortawesome/react-fontawesome": "^0.2.0",
+    "prettier": "^3.1.0",
```
Comment (AnthYou) [inline escapades_frontend/package.json:19]: I would add prettier as a dev dependency instead.

File: escapades_frontend/src/__tests__/components/App.test.tsx [hunk 4]
```diff
@@ -1,9 +1,14 @@
 import { render } from "@testing-library/react";
-import App from "@/components/App";
+import App from "@/pages/App";
```
Comment (AnthYou) [inline escapades_frontend/src/__tests__/components/App.test.tsx:2]: ✨

File: escapades_frontend/src/__tests__/components/App.test.tsx [hunk 5]
```diff
@@ -1,9 +1,14 @@
...
+import { BrowserRouter } from "react-router-dom";
 
 describe("<App />", () => {
   it("should render successfully", () => {
-    const { baseElement } = render(<App />);
+    const { baseElement } = render(
+      <BrowserRouter>
+        <App />
+      </BrowserRouter>
+    );
```
Comment (AnthYou) [inline escapades_frontend/src/__tests__/components/App.test.tsx:11]: Good one ☝️

File: escapades_frontend/src/__tests__/components/Home.test.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    const { baseElement } = render(<Home />);
+    expect(baseElement).toBeTruthy();
+  });
+
+  it("should have title message visible", () => {
+    render(<Home />);
+    const title = screen.queryByText(/Your next trip is waiting for you/);
+    expect(title).toBeVisible();
+  });
+});
```
Comment (AnthYou) [inline escapades_frontend/src/__tests__/components/Home.test.tsx:15]: ✨

File: escapades_frontend/src/components/App.tsx
Comment (AnthYou) [inline escapades_frontend/src/components/App.tsx:1]: I would have let the `App.tsx` file in this folder.

File: escapades_frontend/src/components/HomeImg/CoverCard.module.scss [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.imgContainer {
+  @apply relative w-screen h-[33vh] sm:h-[66vh] bg-cover;
+
+  &__title {
+    @apply absolute top-1/4 sm:top-1/3 left-1/2 z-[9998] text-white text-base sm:text-3xl xl:text-5xl;
+    transform: translate(-50%, 0);
+  }
+}
```
Comment (AnthYou) [inline escapades_frontend/src/components/HomeImg/CoverCard.module.scss:8]: Nice here ✨

However, for the naming convention, it is a good practice to name the folder with the same name as the component.
Here I think we may call it HomeBanner, what do you think?

File: escapades_frontend/src/components/HomeImg/CoverImg.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { FC } from "react";
+import classes from "./CoverCard.module.scss";
+
+type CoverCardProps = {
+  url: string;
+  text: string;
+};
```
Comment (AnthYou) [inline escapades_frontend/src/components/HomeImg/CoverImg.tsx:7]: As discussed, I would prefer to use interface here. However it works as well.

File: escapades_frontend/src/components/HomeImg/CoverImg.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { FC } from "react";
+import classes from "./CoverCard.module.scss";
+
+type CoverCardProps = {
+  url: string;
```
Comment (AnthYou) [inline escapades_frontend/src/components/HomeImg/CoverImg.tsx:5]: Maybe we could have been a bit more explicit here: I suggest `imageUrl`

File: escapades_frontend/src/components/Navbar/Navbar.module.scss (hunk 1 above)
Comment (AnthYou) [reply escapades_frontend/src/components/Navbar/Navbar.module.scss:13]: Have you tried this pattern:

```scss
// Example Module Name: myModule

.myModule {
  // Module styles

  &__block {
    // Block styles

    &--modifier {
      // Modifier styles
    }

    &__element {
      // Element styles

      &--modifier {
        // Modifier styles
      }
    }
  }

  &__anotherBlock {
    // Another block styles
  }
}
```

File: escapades_frontend/src/components/Navbar/Navbar.module.scss (hunk 2 above)
Comment (AnthYou) [reply escapades_frontend/src/components/Navbar/Navbar.module.scss:26]: I let you check my comment above

File: escapades_frontend/src/components/Navbar/Navbar.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import { FC, useState } from "react";
+import { Link, useMatch, useResolvedPath } from "react-router-dom";
+import classes from "./Navbar.module.scss";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBars, faXmark } from "@fortawesome/free-solid-svg-icons";
+
+type NavBarLinkProps = {
+  to: string;
+  children: string;
+};
```
Comment (AnthYou) [inline escapades_frontend/src/components/Navbar/Navbar.tsx:10]: Same remark here about type vs interface

File: escapades_frontend/src/main.tsx [hunk 11]
```diff
@@ -1,10 +1,13 @@
...
-ReactDOM.createRoot(document.getElementById('root')!).render(
+ReactDOM.createRoot(document.getElementById("root")!).render(
   <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-)
+    <BrowserRouter>
+      <App />
+    </BrowserRouter>
+  </React.StrictMode>
```
Comment (AnthYou) [inline escapades_frontend/src/main.tsx:12]: ✨

File: escapades_frontend/src/pages/App.tsx
Comment (AnthYou) [inline escapades_frontend/src/pages/App.tsx:1]: So I would move this file to /components as it is not a page.

File: escapades_frontend/src/pages/Register.tsx [hunk 12]
```diff
@@ -0,0 +1,6 @@
+const Register = () => (
+  <h1>Register</h1>
+)
+
+export default Register;
+
```
Comment (AnthYou) [inline escapades_frontend/src/pages/Register.tsx:6]: Careful there is one extra whiteline here

**Review by AnthYou — CHANGES_REQUESTED** [review]
Nice work overall. ✨
Some little issues to fix (see my comments).

Careful, there are two issues on the responsive view:
Screenshot 2023-12-12 at 16 18 18
- The modal of the navbar is not centered, it is cut on the right side of the screen
- The title is not centered when switching to mobile
- Do not forget to add a screenshot of the responsive view in the PR description if possible/appropriate.

🙏

File: escapades_frontend/package.json (hunk 3 above)
Comment (mathieu-valery) [reply escapades_frontend/package.json:19]: nice ! I forgot about devDependencies  I forgot about devDependencies

File: escapades_frontend/src/components/HomeImg/CoverImg.tsx (hunk 8 above)
Comment (mathieu-valery) [reply escapades_frontend/src/components/HomeImg/CoverImg.tsx:7]: as you want, they are very similar anyway
here's a good article on this topic https://www.totaltypescript.com/type-vs-interface-which-should-you-use

File: escapades_frontend/src/components/Navbar/Navbar.module.scss (hunk 1 above)
Comment (mathieu-valery) [reply escapades_frontend/src/components/Navbar/Navbar.module.scss:13]: thks but when I try to call the modifier classes in my component ex: `classes.Navbar__links--open` it doesnt work and I have a parsing error :/

**Comment by mathieu-valery** [comment]
- The modal of the navbar is not centered, it is cut on the right side of the screen
_you're right ! I've found a solution_
- The title is not centered when switching to mobile
_pretty sure it is centered but it missed a max content width_
- Do not forget to add a screenshot of the responsive view in the PR description if possible/appropriate.
_oops yep I forgot, ils done ! _

File: escapades_frontend/src/components/HomeImg/CoverCard.module.scss (hunk 7 above)
Comment (mathieu-valery) [reply escapades_frontend/src/components/HomeImg/CoverCard.module.scss:8]: lets go for HomeBanner

File: escapades_frontend/src/__tests__/components/Home.test.tsx
Comment (AnthYou) [inline escapades_frontend/src/__tests__/components/Home.test.tsx:1]: I just noticed, but in the `__tests__` folder, please replicate the same architecture than in the files. Here, the `Home.test.tsx` should be in a `pages` folder, to follow the path in the files.

**Review by AnthYou — APPROVED** [review]
It should be good, a last change in the tests folder and let's merge it :)

**Comment by AnthYou** [comment]
> * The modal of the navbar is not centered, it is cut on the right side of the screen
>   _you're right ! I've found a solution_
> * The title is not centered when switching to mobile
>   _pretty sure it is centered but it missed a max content width_
> * Do not forget to add a screenshot of the responsive view in the PR description if possible/appropriate.
>   _oops yep I forgot, its done_

I don't see the responsive screenshot in the PR description?

**Pull request merged**